### ISSUE TITLE
Issue when calling callProcedure

### ISSUE THREAD

**Issue by Shao-Eric** [markdirish/node-odbc#102]
Hi,

(Am on a Windows 10 OS, Node 10.16.0, DB: MSSQL)

I was able to get 1000 rows for the below query, which means I have successfully connected to the DB
const result = await connection.query(`SELECT top 1000 * from tableName`) 

The issue happens when I try to call the procedure: 
 const procResult= await connection.callProcedure(null, "dbo", 'sp_procname', ['06-30-2020', 0, 223906, 1, 0], (error, res) => {
                if(error) console.log('error', error)
  // "Error: [odbc] The number of parameters the procedure expects and and the number of passed parameters is not equal
 })

I also tried odbc version 2.2.0 and the latest 2.3.0, do you see what I have done wrong here?

If there is an issue with callProcedure, can I use query or statement to call procedure instead? If yes, could you please provide an example? I tried connection.query("exec sp_procnam 'arg1', arg2, arg3, arg4, arg5") but it didn't work for me.

Thank you so much for your help!

**Comment by markdirish** [comment]
Hi @Shao-Eric,

Looking at your code, the first thing that jumps out at me is that you are using both callback and Promise syntax. First thing I would try is rewriting it like: 

```
try {
  const procResult= await connection.callProcedure(null, "dbo", 'sp_procname', ['06-30-2020', 0, 223906, 1, 0];
  console.log(procResult);
} catch (error) {
  console.error(error);
}
```
I don't think that will _fix_ your issue, since it is probably just defaulting to using the callback and giving you an error from the code that looks correct.

The error you are getting means that the application called `SQLProcedureColumns` for your given procedure, which returned the number of columns/parameters needed to call the procedure. Then you sent in 5 values, but this didn't match the number of columns it was expected from the previous call. Are you sure that it takes 5 values?

To check what it is returning, you can add `DEBUG` to the `defines` list in `binding.gyp`, rebuild using `npm install`, then rerun your command. It might be hard to parse, but if you post it here I can see better what is going on (or if you feel more comfortable, you can email me the output at mirish@ibm.com)

**Comment by Shao-Eric** [comment]
All is good now, thank you so much for following up on this, I am going to close it
